source-manager/src/main/java/com/jointrivial/sourcemanager/nordigen/web/controller/NordigenController.java

```diff
@@ -59,15 +65,26 @@ public NordigenController(CountryService countryService, BankService bankService
     public ResponseEntity<AuthorizationLinkViewModel> getAuthorizationLink(
             @RequestHeader("User-Token") String userToken,
             @RequestHeader("Bank-Id") String bankId,
+            @RequestHeader("Bank-Name") String bankName,
             @RequestHeader("Redirect-Link-Prefix") String redirectLinkPrefix) throws IOException, InterruptedException {
 
         // TODO check if userToken is valid with a request to account microservice
-
         AuthorizationLinkViewModel authorizationLinkViewModel =
-                this.sourceIdentifierService.getAuthorizationLink(userToken, bankId, redirectLinkPrefix);
+                this.nordigenConnectionIdService.getAuthorizationLink(userToken, bankId, bankName, redirectLinkPrefix);
 
         // TODO add exc handler
 
         return new ResponseEntity<>(authorizationLinkViewModel, HttpStatus.OK);
     }
+
+    @CrossOrigin(origins = "http://localhost:3000")
+    @GetMapping(value = "/verify_requisition",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public HttpStatus verifyRequisition(
+            @RequestHeader("User-Token") String userToken,
+            @RequestHeader("Reference-Id") String referenceId) throws IOException, InterruptedException {
+
+        return this.nordigenConnectionIdService.verifyRequisition(referenceId,userToken);
```

Comment:
It is much better now, in my humble opinion. Though, too much abstraction :D Can we at least have the return of the verifyRequisition method in a separate variable so that we can clearly see what is eventually returned from the controller?